products.json

```diff
@@ -151,7 +171,19 @@
             2500,
             9000
           ]
-        }
+        },
+        "upgrades": [
+          [
+            2,
+            80,
+            {
+              "temperature_range": [
+                1500,
+                9000
+              ]
+            }
+          ]
```

Comment:
This format makes it harder to parse in a typed language (like go), because the items in this list are of different types.

I would suggest to change this to a map instead, like:

```json
"upgrades": [
  {
    "major": 2,
    "minor": 80,
    "values": {
      "temperature_range": [
        1500,
        9000
      ]
    }
  }
]
```

Reply:
that makes sense, I'll do that. Thanks!

Reply:
fixed :)